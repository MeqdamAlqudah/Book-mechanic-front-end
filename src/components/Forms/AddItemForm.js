import axios from 'axios';
import { useSelector } from 'react-redux';
import store from '../../redux/configureStore';

const AddItemForm = () => {
  const POST_CARS_DATA = 'POST_CARS_DATA';
  const user = useSelector((el) => el.userReducer.find((item) => item === 'currentUser'));
  const submitHandler = (event) => {
    event.preventDefault();
    axios({
      method: 'post',
      url: 'http://127.0.0.1:3000/api/v1/cars',
      data: {
        brand: event.target[0].value,
        model: event.target[1].value,
        registration: event.target[2].value,
        photo: event.target[3].value,
        user_id: user.id,
      },
      headers: {
        'Content-Type': 'application/json',

      },
    }).then((res) => store.dispatch(
      { type: POST_CARS_DATA, newitem: res.data },
    )).catch((error) => {
      throw Error(error);
    });
  };

  return (
    <form onSubmit={submitHandler}>
      <input type="text" name="brand" placeholder="Car Brand" />
      <input type="text" name="model" placeholder="Car Model" />
      <input type="text" name="registration" placeholder="Car Registration" />
      <input type="text" name="photo" placeholder="Car Photo Url" />
      <button type="submit">Submit</button>
    </form>
  );
};

export default AddItemForm;
