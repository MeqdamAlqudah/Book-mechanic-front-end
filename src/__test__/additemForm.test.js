import * as React from 'react';
import { Provider } from 'react-redux';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import store from '../redux/configureStore';
import AddItemForm from '../components/Forms/AddItemForm';

const renderer = require('react-test-renderer');

global.React = React;
describe('Test CarDetail', () => {
  test('test CarDetail rendering ', () => {
    const useSelector = jest.fn(() => 1);
    const AxiosWrapper = jest.fn(() => Promise.resolve({
      json: () => Promise.resolve({
        brand: 'BMW',
        model: 'Gt-200',
        photo: 'http/photo',
        registration: '232423-sad',
      }),
    }));

    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <AddItemForm />
        </Router>
      </Provider>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
