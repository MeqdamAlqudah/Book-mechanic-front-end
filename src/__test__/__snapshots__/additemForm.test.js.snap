// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CarDetail test CarDetail rendering  1`] = `
<div
  className="main"
>
  <form
    onSubmit={[Function]}
  >
    <input
      name="brand"
      placeholder="Car Brand"
      type="text"
    />
    <input
      name="model"
      placeholder="Car Model"
      type="text"
    />
    <input
      name="registration"
      placeholder="Car Registration"
      type="text"
    />
    <input
      name="photo"
      placeholder="Car Photo Url"
      type="text"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
